lightning/src/util/events.rs

```diff
@@ -1419,22 +1443,15 @@ impl MaybeReadable for Event {
 			25u8 => {
 				let f = || {
 					let mut prev_channel_id = [0; 32];
-					let mut failed_next_destination_opt = None;
+					let mut failed_next_destination_opt = UpgradableRequired(None);
```

Comment:
Can `UpgradableRequired` be hidden inside the macros like we do for `WithoutLength`?

Reply:
Could do, but then it wouldn't have a purpose anymore. The purpose of `RequiredWrapper` + `UpgradableRequired` is that we can treat fields as non-`Option`s in `_decode_tlv`, i.e. do `$field = Readable::read(..)?` instead of `$field = Some(Readable::read(..)?)`. We'd have to do the latter if we hid it.

Reply:
We just made it uniform with the `required` type in this PR, so we could change both of them, I don't mind, but I think it should be done in follow-up.